### DISCUSSION TITLE
Always set `stream: 'web'`?

### DISCUSSION THREAD

**Discussion by brillout** [vikejs/bati#303] [Misc]
AFAIK Bati apps don't work with Node.js streams (unless the user removes the Universal Adapter adapter).

How about Bati always sets `stream` to `'web'`? In order to avoid https://github.com/vikejs/vike-react/issues/129.

**Comment by jordyalcides** [comment]
It seems to me that `stream: 'web'` causes less trouble for most use cases, but this might be a bias due to my own use cases

**Comment by magne4000** [comment]
@brillout This would enable `stream` mode by default, and I'm still not convinced I want it by default. For instance, `react-streaming` disables streaming mode for bots, but that's not the case for Solid and Vue. Streaming also does not work if there is no client-side JavaScript enabled, like with simple fetch-based testing.

Currently the `stream` config property is used to toggle streaming AND configure it, but we should probably have 2 different properties for that: `stream: boolean` and `streamMode: 'web' | 'node' | 'auto'`. Perhaps Vike CLI could also solve this activation vs. configuration problem?

**Reply by brillout** [reply]
:+1: Makes sense. How about we enable it by default only for React? All React extensions need it (`vike-react-query`, `vike-react-apollo`, `vike-react-zustand`, `vike-react-telefunc`, and more). These extensions will be increasingly important (I forsee most React apps to use at least one of them) so I think it's worth it to enable it.

> Currently the `stream` config property is used to toggle streaming AND configure it, but we should probably have 2 different properties for that: `stream: boolean` and `streamMode: 'web' | 'node' | 'auto'`.

~~Bati is the only use case I can think of for creating an extra setting like this. So I'm a bit hesitant.~~

Actually, once we create server Vike extensions (`vike-express`, `vike-hono`, ...), we can make `streamMode` a semi-documented setting and let these extensions set it.

I'm :+1: For a new setting `streamType`. (I think it's a slightly better fitting name, but feel free to disagree.)

**Reply by brillout** [reply]
> How about we enable it by default only for React? 

We don't need this if we create a new setting `streamType`. (Because the React extensions already set `streamIsRequired: true`.)
